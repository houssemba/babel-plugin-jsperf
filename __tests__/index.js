/*eslint no-undef: "error"*/
/*eslint-env node*/
const pluginTester = require('babel-plugin-tester');
const jsperfPlugin = require('../src/index');

pluginTester({
  plugin: jsperfPlugin,
  snapshot: true,
  tests: [{
    title: 'function without return statement',
    code: `
      // jsperf
      function todo() {
        console.log('toto');
      }
    `
  }, {
    title: 'function with return statement',
    code: `
      // jsperf
      function foo() {
        const bar = 'bar';
        return bar;
      }
    `
  }, {
    title: 'arrow function',
    code: `
      const foo = /* jsperf */ () => 'test';
    `
  }, {
    title: 'clazz method',
    code: `
      class clazz {
        // jsperf
        method() {
          console.log('class method !');
        }
      }
    `
  }, {
    title: 'function without comment',
    code: `
      function fun() {
         console.log('Do not touch !');
      }
    `,
    snapshot: false
  }]
});
