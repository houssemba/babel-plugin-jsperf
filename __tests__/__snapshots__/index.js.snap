// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arrow function 1`] = `
"
const foo = /* jsperf */ () => 'test';

      ↓ ↓ ↓ ↓ ↓ ↓

const foo = /* jsperf */function () {
  console.time('foo');
  var _returnValue = 'test';
  console.timeEnd('foo');
  return _returnValue;
};
"
`;

exports[`clazz method 1`] = `
"
class clazz {
  // jsperf
  method() {
    console.log('class method !');
  }
}

      ↓ ↓ ↓ ↓ ↓ ↓

class clazz {
  // jsperf
  method() {
    console.time('function');

    console.log('class method !');
    console.timeEnd('function');
  }
}
"
`;

exports[`function with return statement 1`] = `
"
// jsperf
function foo() {
  const bar = 'bar';
  return bar;
}

      ↓ ↓ ↓ ↓ ↓ ↓

// jsperf
function foo() {
  console.time('foo');

  const bar = 'bar';
  var _returnValue = bar;
  console.timeEnd('foo');
  return _returnValue;
}
"
`;

exports[`function without return statement 1`] = `
"
// jsperf
function todo() {
  console.log('toto');
}

      ↓ ↓ ↓ ↓ ↓ ↓

// jsperf
function todo() {
  console.time('todo');

  console.log('toto');
  console.timeEnd('todo');
}
"
`;
